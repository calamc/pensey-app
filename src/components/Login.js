import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/authentication'


// LOGIN COMPONENT { GOOGLE AUTH }
export const Login = ({ startLogin }) => (
    <div>
        <button onClick={startLogin}>Login</button>
    </div>
);

// STATELESS COMPONENT SO DOESNT NEED ACCESS TO ANYTHING STATE RELATED
// SO NO MAPSTATE TOPROPS

// DISPATCH Login
const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

// CONNECT REDUX
// UNDEFINED WHEN NO STATE ***
export default connect(undefined, mapDispatchToProps)(Login);