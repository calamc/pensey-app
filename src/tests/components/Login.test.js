import React from 'react'
import { shallow } from 'enzyme'

import { Login } from '../../components/Login.js';

test('Login component rendering correctly?', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
});