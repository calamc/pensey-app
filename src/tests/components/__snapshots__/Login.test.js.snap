// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component rendering correctly? 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
